src/store/reducer.js

```diff
@@ -0,0 +1,52 @@
+
+const initialState = {
+    cart : [],
+    totalPrice: 0
+};
+
+const reducer = (state = initialState, action) => {
+
+    switch ((action.type)) {
+        case 'addProduct':
+            let newProduct = {id:action.id, name:action.name, price:action.price, description:action.description,
+                                brand:{id:action.brandId, name:action.brandName}, size:action.size, camera:action.camera, 
```

Comment:
There is no need to keep every detail of the product in the cart. id and amount is enough.